userinterface/planui.cpp

```diff
@@ -230,8 +230,54 @@ void PlanUI::on_generateRouteButton_clicked()
     mRouteGeneratorUI->show();
 }
 
-void PlanUI::on_Reverse_clicked()
+void PlanUI::on_reverseButton_clicked()
 {
     mRoutePlanner->reverseCurrentRoute();
 }
 
+void PlanUI::on_appendButton_clicked()
+{
+    bool ok;    // set to false if user presses Cancel (or "x")
+    int index = QInputDialog::getInt(this, tr("Append to..."),
+                                     tr("Select route to append:"), 1,
+                                     1, mRoutePlanner->getNumberOfRoutes(), 1, &ok);
+
+    if(ok && index != mRoutePlanner->getCurrentRouteIndex()+1)    // ignore if trying to merge with itsef
+    {
+        mRoutePlanner->appendToRoute(index-1);  // -1 to match QList indexes
+
+        ui->currentRouteSpinBox->setValue(mRoutePlanner->getCurrentRouteIndex() + 1);
+        ui->currentRouteSpinBox->setMaximum(mRoutePlanner->getNumberOfRoutes());
+        ui->currentRouteSpinBox->setSuffix(" / " + QString::number(mRoutePlanner->getNumberOfRoutes()));
+    }
+}
+
+void PlanUI::on_splitButton_clicked()
+{
+    if(mRoutePlanner->getCurrentRoute().size() < 2)
```

Comment:
I would rather see a solution that enables/disables the buttons depending on the state of mRoutePlannerModule, e.g.:
```
connect(mRoutePlanner.get(), &RoutePlannerModule::requestRepaint, [this](){
           ui->splitButton->setEnabled(mRoutePlanner->getCurrentRoute().size() > 1);
});
```